packages/node/src/dev-server.ts

```diff
@@ -146,28 +146,35 @@ async function createEdgeEventHandler(
     ${userCode};
 
     addEventListener('fetch', async (event) => {
-      let serializedRequest = await event.request.text();
-      let requestDetails = JSON.parse(serializedRequest);
+      try {
+        let serializedRequest = await event.request.text();
+        let requestDetails = JSON.parse(serializedRequest);
 
-      let body;
+        let body;
 
-      if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
-        body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
-      }
+        if (requestDetails.method !== 'GET' && requestDetails.method !== 'HEAD') {
+          body = Uint8Array.from(atob(requestDetails.body), c => c.charCodeAt(0));
+        }
 
-      let requestUrl = requestDetails.headers['x-forwarded-proto'] + '://' + requestDetails.headers['x-forwarded-host'] + requestDetails.url;
+        let requestUrl = requestDetails.headers['x-forwarded-proto'] + '://' + requestDetails.headers['x-forwarded-host'] + requestDetails.url;
 
-      let request = new Request(requestUrl, {
-        headers: requestDetails.headers,
-        method: requestDetails.method,
-        body: body
-      });
+        let request = new Request(requestUrl, {
+          headers: requestDetails.headers,
+          method: requestDetails.method,
+          body: body
+        });
 
-      event.request = request;
+        event.request = request;
 
-      let edgeHandler = module.exports.default;
-      let response = edgeHandler(event.request, event);
-      return event.respondWith(response);
+        let edgeHandler = module.exports.default;
+        let response = await edgeHandler(event.request, event);
+
+        return event.respondWith(response);
+      } catch (error) {
+        event.respondWith(new Response(error.stack, {
+          status: 500
+        }));
```

Comment:
We handle errors within the `edge-runtime` code so that we don't trigger `edge-runtime`'s error handling. It was doing some extra logging we couldn't control. This gives us direct control of error handling in the response.